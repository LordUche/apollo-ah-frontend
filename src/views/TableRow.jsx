import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { getDisplayDate } from '../utils/helpers';


export default function TableRow({ item }) {
  const date = getDisplayDate(item.createdAt);
  return (
    <tr>
      <td className="align-middle">
        <p className="mb-n1 font-weight-bold">{item.title}</p>
        <small>{date.long}</small>
        <div className="dropdown d-inline-block">
          <button className="btn dropdown-toggle btn-unfocused" type="button" id="dropdownMenuButton" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false" />
          <div className="dropdown-menu" aria-labelledby="dropdownMenuButton">
            <Link className="dropdown-item" to={`/article/${item.slug}`} target="_blank">View article</Link>
            <Link className="dropdown-item" to={`/user/create-article/${item.slug}`} target="_blank">Edit</Link>
          </div>
        </div>
      </td>
      <td className="text-center align-middle">{item.comments}</td>
      <td className="text-center align-middle">{item.claps}</td>
    </tr>
  );
}
TableRow.propTypes = {
  item: PropTypes.object.isRequired,
};
