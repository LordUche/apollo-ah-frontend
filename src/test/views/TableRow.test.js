import React from 'react';
import { shallow } from 'enzyme';
import TableRow from '../../views/TableRow';

describe('<TableRow>', () => {
  it('should render', () => {
    const wrapper = shallow(<TableRow item={{}} />);
    expect(wrapper).toMatchSnapshot();
  });
});
