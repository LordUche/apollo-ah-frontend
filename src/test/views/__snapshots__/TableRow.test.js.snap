// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableRow> should render 1`] = `
<tr>
  <td
    className="align-middle"
  >
    <p
      className="mb-n1 font-weight-bold"
    />
    <small>
      Invalid Date
    </small>
    <div
      className="dropdown d-inline-block"
    >
      <button
        aria-expanded="false"
        aria-haspopup="true"
        className="btn dropdown-toggle btn-unfocused"
        data-toggle="dropdown"
        id="dropdownMenuButton"
        type="button"
      />
      <div
        aria-labelledby="dropdownMenuButton"
        className="dropdown-menu"
      >
        <Link
          className="dropdown-item"
          replace={false}
          target="_blank"
          to="/article/undefined"
        >
          View article
        </Link>
        <Link
          className="dropdown-item"
          replace={false}
          target="_blank"
          to="/user/create-article/undefined"
        >
          Edit
        </Link>
      </div>
    </div>
  </td>
  <td
    className="text-center align-middle"
  />
  <td
    className="text-center align-middle"
  />
</tr>
`;
